libbpf-tools/vfsstat.c

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+//
+// Based on vfsstat(8) from BCC by Brendan Gregg
+#include <sys/resource.h>
+#include <argp.h>
+#include <unistd.h>
+#include <time.h>
+#include <bpf/bpf.h>
+#include "vfsstat.h"
+#include "vfsstat.skel.h"
+
+const char *argp_program_version = "vfsstat 0.1";
+const char *argp_program_bug_address = "<bpf@vger.kernel.org>";
+static const char argp_program_doc[] =
+	"\nvfsstat: Count some VFS calls\n"
+	"\n"
+	"EXAMPLES:\n"
+	"    vfsstat      # interval one second\n"
+	"    vfsstat 5 3  # interval five seconds, three output lines\n";
+static char args_doc[] = "[interval [count]]";
+
+static const struct argp_option opts[] = {
+	{ "verbose", 'v', NULL, 0, "Verbose debug output" },
+	{},
+};
+
+static struct env {
+	bool verbose;
+	int count;
+	int interval;
+} env = {
+	.interval = 1,	/* once a second */
+};
+
+static error_t parse_arg(int key, char *arg, struct argp_state *state)
+{
+	long interval;
+	long count;
+
+	switch (key) {
+	case 'v':
+		env.verbose = true;
+		break;
+	case ARGP_KEY_ARG:
+		switch (state->arg_num) {
+		case 0:
+			interval = strtol(arg, NULL, 10);
```

Comment:
forgot to mention in first round, you have to set errno to zero here, otherwise you can erroneously detect errno != 0 from some previous invocation.

Reply:
Thanks @anakryiko, fixed here and below.